### PR TITLE
Create exported namespace package 

### PR DESCRIPTION
**Pull request by jasonBirchall**
After importing this package you can call the function
`namespace.ChangedInPR` which will return all the namespace names
changed in a Pull request.

This is useful for a number of reasons, but more specifically it allows
us to deduplicate the namespace and perform meta actions like ensuring
the PR owner is allowed to make changes to that namespaces.

### CODE REVIEW

File: pkg/namespace/namespace.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+// It queries the GitHub API for all changes made in a PR. If the PR contains changes to a namespace
+// it returns a deduplicated slice of namespace names.
+func ChangedInPR(branchRef, token, repo, owner string) ([]string, error) {
+	if token == "" {
+		return nil, errors.New("You must have a valid GitHub token.")
+	}
+
+	client := githubClient(token)
+
+	str := strings.Split(branchRef, "/")
```
Comment (poornima-krishnasamy) [inline pkg/namespace/namespace.go:27]: Can you please add an example comment on how branchRef would look like so it is easy to understand the fornat

Comment (jasonBirchall) [reply pkg/namespace/namespace.go:27]: good thinking. Adding now

File: pkg/namespace/namespace.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	repos, _, _ := client.PullRequests.ListFiles(context.Background(), owner, repo, prId, nil)
+
+	var namespaceNames []string
+	for _, repo := range repos {
+		fmt.Println(*repo.Filename)
+		if strings.Contains(*repo.Filename, "live") {
+			// namespaces filepaths are assumed to come in
+			// the format: namespaces/live-1.cloud-platform.service.justice.gov.uk/<namespaceName>
+			s := strings.Split(*repo.Filename, "/")
+			namespaceNames = append(namespaceNames, s[3])
```
[the remark this answers is not in the document]
Comment (jasonBirchall) [reply pkg/namespace/namespace.go:42]: absolutely spot on. Thanks for catching.

[the remark this answers is not in the document]
Comment (jasonBirchall) [reply pkg/namespace/namespace.go]: a7c5395

File: pkg/namespace/namespace.go (hunk 1 above)
Comment (jasonBirchall) [reply pkg/namespace/namespace.go:29]: a7c5395

**Review by poornima-krishnasamy — APPROVED** [review]

**Pull request merged**